### PR TITLE
Support more mouse buttons

### PR DESCRIPTION
**Pull request by kraflab**
Addresses https://github.com/coelckers/prboom-plus/issues/154

I just brought over the crispy implementation.

### CODE REVIEW

File: prboom2/src/m_menu.c [hunk 1]
```diff
@@ -4548,7 +4548,7 @@ dboolean M_Responder (event_t* ev) {
       // to where key binding can eat it.
 
       if (setup_active && set_keybnd_active)
-  if (ev->data1&4 || ev->data1&8 || ev->data1&16)
+  if (ev->data1&(4 + 8 + 16 + 32 + 64 + 128))
```
Comment (fabiangreffrath) [inline prboom2/src/m_menu.c:4551]: How about `if (ev->data1 >> 2)`?

File: prboom2/src/m_menu.c [hunk 2]
```diff
@@ -5214,6 +5214,12 @@ dboolean M_Responder (event_t* ev) {
       ch = 3;
     else if (ev->data1 & 16)
       ch = 4;
+    else if (ev->data1 & 32)
```
Comment (fabiangreffrath) [inline prboom2/src/m_menu.c:5217]: How about this?
https://github.com/fabiangreffrath/woof/blob/f0c5d0733f005b97cf7e69055573ef49914df7e4/Source/m_menu.c#L5034
https://github.com/fabiangreffrath/woof/blob/f0c5d0733f005b97cf7e69055573ef49914df7e4/Source/m_menu.c#L4354

**Review by fabiangreffrath — APPROVED** [review]

File: prboom2/src/m_menu.c [hunk 3]
```diff
@@ -5204,17 +5221,7 @@ dboolean M_Responder (event_t* ev) {
...
-    else if (ev->data1 & 2)
-      ch = 1;
-    else if (ev->data1 & 4)
-      ch = 2;
-    else if (ev->data1 & 8)
-      ch = 3;
-    else if (ev->data1 & 16)
-      ch = 4;
-    else
+    if ((ch = GetButtons(MAX_MOUSE_BUTTONS, ev->data1)) == -1)
```
Comment (kraflab) [inline prboom2/src/m_menu.c:5224]: I ended up in the wrong place somehow 😵

File: prboom2/src/m_menu.c (hunk 2 above)
Comment (kraflab) [reply prboom2/src/m_menu.c]: done

File: prboom2/src/m_menu.c (hunk 1 above)
Comment (kraflab) [reply prboom2/src/m_menu.c]: done

**Comment by kraflab** [comment]
Also fixes https://github.com/coelckers/prboom-plus/issues/78

**Comment by fabiangreffrath** [comment]
Top, thanks!

**Pull request merged**